src/System.Management.Automation/engine/GetCommandCommand.cs

```diff
@@ -1306,7 +1306,9 @@ private bool IsCommandMatch(ref CommandInfo current, out bool isDuplicate)
                     {
                         // If the matching command was an alias, then use the resolved command
                         // instead of the alias...
-                        current = ai.ResolvedCommand;
+                        current = ai.ResolvedCommand == null ?
+                            CommandDiscovery.LookupCommandInfo(ai.UnresolvedCommandName, this.MyInvocation.CommandOrigin, this.Context) :
+                            ai.ResolvedCommand;
                         if (current == null)
```

Comment:
Seems this will be dead code now? Is that correct?

Reply:
@ChrisLGardner ping..

Reply:
No, doing some testing and it's there to resolve the command name if the module the alias is in hasn't been loaded yet and then that can be used to get the syntax. Without it you get the below behaviour:

```
PS> gcm del -syn
Remove-Item
```

with it you get this:

```
PS>  gcm del -syn
del (alias) -> Remove-Item

del [-Path] <string[]> [-Filter <string>] [-Include <string[]>] [-Exclude <string[]>] [-Recurse] [-Force] [-Credential <pscredential>] 

del -LiteralPath <string[]> [-Filter <string>] [-Include <string[]>] [-Exclude <string[]>] [-Recurse] [-Force] [-Credential <pscredential>] [-WhatIf] [-Confirm] [-Stream <string[]>] [<CommonParameters>]
```

Reply:
@ChrisLGardner I think @adityapatwardhan is referring to the line 1312 `if (current == null)` code path, which shouldn't be reached anymore since you're checking for null and accounting for that directly. If whatever's in that code path isn't reachable anymore, it should be removed. 🙂 

Also, this condition with the null can be simplified:

```csharp
current = ai.ResolvedCommand ?? CommandDiscovery.LookupCommandInfo(
    ai.UnresolvedCommandName,
    this.MyInvocation.CommandOrigin,
    this.Context);
```

Reply:
Good point, I'll get that fixed now.

Reply:
Looking at the failure in the tests there's a reason that check is there. It can sometimes return null from LookupCommandInfo. In the case for the tests (and locally) it is because `spvm` has a null `ResolvedCommand` and `UnresolvedCommandName`, probably in my case because I've apparently got both 1.1 and 2.0.0.0 of Hyper-V installed and I'd guess the AzDOS agents are similar.

Reply:
Might want to add a comment explaining that so nobody tries to remove that later 🙂